tombstones.go

```diff
@@ -156,30 +157,55 @@ func readTombstones(dir string) (memTombstones, error) {
 	return stonesMap, nil
 }
 
-type memTombstones map[uint64]Intervals
+type memTombstones struct {
+	mts map[uint64]Intervals
+	mtx sync.RWMutex
+}
+
+func newEmptyMemTombstones() *memTombstones {
```

Comment:
I think it is an easier way create an memTombstones object.

Reply:
if you prefer it this way than it can  be simplified a bit. Also the name can be just `newMemTombstones`
```
func newMemTombstones() *memTombstones {
     return &memTombstones{mts: make(map[uint64]Intervals)}
}
```